modules/api/common_types.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
```

Comment:
I think we could also add the Debug type and maybe a generic PasswordSelector struct?

Reply:
yep i think both makes sesne to add.

Reply:
Debug: I think each Service CRD will have a Debug.Service like parameter to stop the container before starting the service binary. But not each service will have Debug.DBSync as not each service (at least in nova) will run dbsync on the DB it uses. The APIDB in nova is shared between the NovaAPI, NovaScheduler, NovaConductor. Btw, if we are talking about Debug, I see PreserveJobs also something that belongs to Debug. But again maybe not all service will run Jobs.  Bottom line I'm not sure that Debug is generic enough to pulled out. 

PasswordSelector: That is definitely service specific, the default values there are different for each service type.

Reply:
the password selector should liekly just be a reference to a secreate containing all the password that service needs

Reply:
@SeanMooney: The PasswordSelector does not select a Secret. The Secret is selected via the Secret string field via the name of the Secret.  The PasswordSelector defines the field names the operator should look at in the Secret to find certain things. For me this is a bit too much indirection. I would hardcode the field names. But maybe @dprince or @stuggi has some good examples when this extra indirection is useful. Anyhow the current PasswordSelector cannot be pulled out to the generic Service struct as the default values in it are service specific.

Debug is setting debug mode for the operator behavior not Debug on the sevice binary.